packages/ui/primitives/stepper.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useEffect, useState } from 'react';
```

Comment:
This is a pretty cool implementation, would be nice to potentially use context here rather than cloneElement so we don't have to concern ourselves with injected props which can be hard to reason about.